sgkit/distance/api.py

```diff
@@ -60,6 +62,8 @@ def pairwise_distance(
 
         Omit to let dask heuristically decide a good default. A default can
         also be set globally with the split_every key in dask.config.
+    target
```

Comment:
I think `target` was a good name for this in the context of numba given what it does, but I think `device` is better at our API level.

Reply:
FYI @jeromekelleher on dispatching, I think by accepting this we're effectively throwing our hands up at that problem and leaving it up to users with this solution (ideally they could define arrays on gpu backends with no further code changes).  I'm cool with that for now but I wanted to point it out to you as well.

Reply:
One of the motivations of not using device was: `device` is also a synonym of GPU, isn't it?

Reply:
The new Python array API uses "device" in this context: https://data-apis.org/array-api/latest/design_topics/device_support.html#device-support. So it seems like the right term to be using here.

Reply:
Ok, great. Sounds good.